sbnana/CAFAna/Systs/EnergySysts.h

```diff
@@ -0,0 +1,240 @@
+//ETW May 2020 Fresh version for SBN
+
+#pragma once
+
+#include "sbnana/CAFAna/Core/ISyst.h"
+#include "sbnana/CAFAna/Core/Utilities.h"
+#include "sbnana/CAFAna/Analysis/ExpInfo.h"
+#include "sbnanaobj/StandardRecord/Proxy/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+
+namespace ana
+{
+
+  enum class EnergyScaleSystTerm {
+    Constant,
+    Sqrt,
+    InverseSqrt
+  };
+
+  enum class EnergyScaleSystParticle {
+    Muon,
+    Hadron,
+    Neutron,
+    EM,
+    ChargedHadron
+  };
+
+  enum class EnergyScaleSystDetector {
+    SBND,
+    MicroBooNE,
+    ICARUS,
+    All
+  };
+
+  template<EnergyScaleSystTerm term, EnergyScaleSystParticle part, EnergyScaleSystDetector detector = EnergyScaleSystDetector::All>
+  class EnergyScaleSyst: public ISyst
+  {
+  public:
+    EnergyScaleSyst(double _uncertainty, std::string name) : 
+      ISyst(name, name), uncertainty(_uncertainty) {}
+    void Shift(double sigma, caf::SRSliceProxy *sr, double& weight) const override
+    {
+      double scale = uncertainty * sigma;
+      auto baseline_cut = [](double baseline){ 
```

Comment:
Does this thing really need to be a lambda? Why not just include this code directly at the top of the Shift() function?